### PR TITLE
Working sort columns

### PR DESCRIPTION
**Pull request by Krzyfho**
In helper added possibility to improve sorting.

### CODE REVIEW

File: app/controllers/expenses_controller.rb [hunk 1]
```diff
@@ -48,4 +49,12 @@ def destroy
...
     params.require(:expense).permit(:name, :price, :paid_on)
   end
+
+  def sort_column
+    Expense.column_names.include?(params[:sort]) ? params[:sort] : "name"
+  end
+
+  def sort_direction
+    %w[asc desc].include?(params[:direction]) ? params[:direction] : "asc"
+  end
```
Comment (hesler) [inline app/controllers/expenses_controller.rb:59]: 👍

File: app/controllers/expenses_controller.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class ExpensesController < ApplicationController
   before_action :authenticate_user!
+  helper_method :sort_column, :sort_direction
 
   def index
-    @expenses = Expense.all
+    @expenses = Expense.order(sort_column + " " + sort_direction)
```
Comment (hesler) [inline app/controllers/expenses_controller.rb:6]: You could also use a hash notation here:
```
Expense.order(sort_column => sort_direction)
```

File: app/views/expenses/index.html.haml [hunk 3]
```diff
@@ -7,11 +7,28 @@
...
+              = I18n.t('expenses.sort.by')
+            %ul.dropdown-menu{"aria-labelledby": "dropdownMenuButton1"}
+              %li
+                = sortable "name"
+              %li
+                = sortable "price"
+              %li
+                = link_to t('expenses.sort.oldest'), { sort: "paid_on", direction: "asc" }, { class: "btn" }
+              %li
+                = link_to t('expenses.sort.newest'), { sort: "paid_on", direction: "desc" }, { class: "btn" }
```
Comment (hesler) [inline app/views/expenses/index.html.haml:23]: Why not `sortable` here? :P

Comment (Krzyfho) [reply app/views/expenses/index.html.haml:23]: I was using two different sort methods and i want your opinion which of theese is better :D

**Review by hesler — APPROVED** [review]

**Pull request merged**